### ISSUE TITLE
CSS HMR errors from Tailwind arbitrary value classes

### ISSUE THREAD

**Issue by miosenpai** [nuxt/nuxt#26454]
### Environment

```
------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.11.1
- CLI Version:      3.11.1
- Nitro Version:    2.9.4
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools, modules, tailwindcss
- Runtime Modules:  @nuxtjs/tailwindcss@6.11.4
- Build Modules:    -
------------------------------
```

### Reproduction

https://stackblitz.com/edit/github-4um3z1-gdby4j

### Describe the bug

When an [arbitrary value Tailwind class with whitespace](https://tailwindcss.com/docs/adding-custom-styles#handling-whitespace) is used (e.g. `font-['Times_New_Roman']`), CSS HMR will throw an error on subsequent hard refreshes.

I had originally reported this issue in https://github.com/nuxt-modules/tailwindcss/issues/815, but @ineshbose later discovered the HMR error occurs even with a manual Tailwind CSS setup (without the module), leading me to forward the issue here. Both of us also suspect this could be a Vite bug, but I have not been able to reproduce the issue in a plain Vite project (e.g. `npm create vue@latest`).

Error Repro:
1. Open the reproduction project above.
2. Edit `@/assets/custom.css` (e.g. change `color: aqua;` to `color: magenta;`), then save the file.
3. Hard refresh the page, an error similar to what's below will be thrown.

```
ERROR  Internal server error: Parse error /home/projects/github-4um3z1/node_modules/tailwindcss/tailwind.css?direct:489:32                                 7:22:34 PM
      at parse$e (/home/projects/github-4um3z1/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:16225:355)
      at handleModuleSoftInvalidation (/home/projects/github-4um3z1/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:53757:27)
      at async getCachedTransformResult (/home/projects/github-4um3z1/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:53562:10)
      at async doTransform (/home/projects/github-4um3z1/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:53535:24)
      at async viteTransformMiddleware (/home/projects/github-4um3z1/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:63458:32)
```

### Additional context

There are some workarounds documented in https://github.com/nuxt-modules/tailwindcss/issues/815#issuecomment-2002459123 that might be helpful in deducing what's actually causing this issue.

**Comment by luc122c** [comment]
Instead of importing the CSS file in the app script, why not include it in the [CSS field](https://nuxt.com/docs/api/nuxt-config#css) in nuxt.config? It seems to resolve the issue: https://stackblitz.com/edit/github-4um3z1-ych9wt?file=nuxt.config.ts

**Comment by miosenpai** [comment]
> Instead of importing the CSS file in the app script, why not include it in the [CSS field](https://nuxt.com/docs/api/nuxt-config#css) in nuxt.config? It seems to resolve the issue: https://stackblitz.com/edit/github-4um3z1-ych9wt?file=nuxt.config.ts

Yes I'm aware of this approach (as discussed by the Tailwind module's developer), but the behavior when CSS is imported in `app.vue` seems like a bug so I've reported it as such.

**Comment by danielroe** [comment]
Super-quickly looked at this and could not reproduce with the following Vue SSR template: https://stackblitz.com/edit/github-e9jfew?file=src%2FApp.vue,src%2Fpages%2FHome.vue,src%2Fcustom.css - in case it's helpful for someone picking this up later.

**Comment by hecktarzuli** [comment]
We get these same errors, but without hard refreshes during HMR.  Doing a hard refresh actually fixes it for us until something is saved again.  What's really weird is this even happens when you're just saving a small change to a composable!

**Comment by miosenpai** [comment]
@danielroe Through a bit of triaging, I have found the following things with regards to this issue:

1. After comparing with an older Nuxt project which didn't have this issue, I tried downgrading Vite using the `package.json` snippet below, and the error did go away. From this, I was able to determine the issue started with Vite 5 (which made more sense after I saw the code which is throwing these errors was added in Vite 5: https://github.com/vitejs/vite/pull/14654).
```json
"overrides": {
  "@nuxt/vite-builder": {
    "vite": "4.5.3"
  }
}
```
2. I also found (as you suspected on stream) it is the `?direct` query that's triggering the error, since a visit to `http://127.0.0.1:3000/_nuxt/node_modules/tailwindcss/tailwind.css` after changing `custom.css` does not trigger any errors.

Not sure how helpful these points will be to tracking down the issue but it's what I have so far, I also tried looking further but got stuck as I'm not familiar with Vite internals.

**Comment by sapphi-red** [comment]
This should be fixed by Vite 5.4.9 👍

**Comment by danielroe** [comment]
@sapphi-red thank you so much for investigating and resolving ❤

**Comment by Xaypanya** [comment]
> This should be fixed by Vite 5.4.9 👍

Thank you for your work! I upgraded, and the issue is now resolved 🎉
